### DISCUSSION TITLE
pgmoneta is unable to run as expected.

### DISCUSSION THREAD

**Discussion by GuChad369** [pgmoneta/pgmoneta#231] [General]
When running the test, I encounter no errors, but `pgmoneta` also fails to run. It simply exits without displaying anything.
Even after rerunning CMake, the issue persists.

<img width="692" alt="Screenshot 2024-03-25 at 5 43 15 PM" src="https://github.com/pgmoneta/pgmoneta/assets/113565433/139ded59-1be5-4e34-9ceb-a5319df97c36">

What could be causing this issue? Any advice would be greatly appreciated.

**Comment by jesperpedersen** [comment]
Configuration ?

**Comment by jesperpedersen** [comment]
Have you `gdb` your way through `main.c` ?

**Comment by GuChad369** [comment]
![image](https://github.com/pgmoneta/pgmoneta/assets/113565433/b37fa5f7-601e-4ee6-bb18-ae23c9101db5)
Why is the configuration suddenly causing problems? I've been using the same configuration each time, and this is the first time I've encountered an issue. It's identical to what I've used before. I've never made any changes to the configuration.

**Reply by jesperpedersen** [reply]
Did you change your master key ?

**Reply by GuChad369** [reply]
No, everything is exactly the same as before. I have a script that reinstalls the code from the updated version, and it has always worked well in helping me test the code. However, today it suddenly seems to be corrupted after several tests.

**Reply by GuChad369** [reply]
The issue today arose because I forgot to stop the previous running instance before reinstalling the new version of the code. However, even after reinstalling everything, it still doesn't seem to work.

**Comment by GuChad369** [comment]
I found it. If encountering this problem, there may be a file named `/tmp/pgmoneta.all.pid`. In such cases, `pgmoneta` won't see the configuration. You'll need to use the command `rm -f /tmp/pgmoneta.all.pid` to delete it.

**Reply by jesperpedersen** [reply]
So, that is an issue that need to be fixed

**Reply by jesperpedersen** [reply]
Or error out with a description

**Reply by GuChad369** [reply]
Is this an issue? This prevents two instances from running on one machine.

I didn't stop the previous instance; I simply reinstalled the code. As a result, the file remained there and was not deleted.  But, I discovered this message in the log.

<img width="974" alt="Screenshot 2024-03-25 at 8 07 02 PM" src="https://github.com/pgmoneta/pgmoneta/assets/113565433/7d43f0b9-dbf8-4072-ac42-d3e5b05f4cf8">

**Reply by Jubilee101** [reply]
Yup, you should check log first if pgmoneta fails to start for some reason. We should perhaps add a reminder for that on pgmoneta start/abort.
